pkg/odo/commonflags/run_on.go

```diff
@@ -0,0 +1,66 @@
+package commonflags
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+
+	"github.com/redhat-developer/odo/pkg/odo/cmdline"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+const (
+	// RunOnFlagName is the name of the flag allowing user to specify target platform
+	RunOnFlagName = "run-on"
+	RunOnCluster  = "cluster"
+	RunOnPodman   = "podman"
+	RunOnDefault  = RunOnCluster
+)
+
+// UseRunOnFlag indicates that a command accepts the --run-on flag
+func UseRunOnFlag(cmd *cobra.Command) {
+	if cmd.Annotations == nil {
+		cmd.Annotations = map[string]string{}
+	}
+	cmd.Annotations["runOn"] = "true"
+}
+
+// AddRunOnFlag adds the --run-on output flag to all commands
+// We use "flag" in order to make this accessible throughtout ALL of odo, rather than the
+// above traditional "persistentflags" usage that does not make it a pointer within the 'pflag'
+// package
+func AddRunOnFlag() {
+	flag.CommandLine.String(RunOnFlagName, "", "Specify target platform, supported platforms: cluster(default), podman")
```

Comment:
Same suggestion as above:

```suggestion
	flag.CommandLine.String(RunOnFlagName, "", "Specify target platform, supported platforms: 'cluster' (default), 'podman' (experimental)")
```